ca/ca.go

```diff
@@ -277,7 +280,11 @@ func (ca *CAImpl) newCertificate(domains []string, ips []net.IP, key crypto.Publ
 		}
 	}
 
-	certNotAfter := time.Now().AddDate(5, 0, 0)
+	certNotAfter := time.Now().AddDate(0, 0, ca.certLifetime)
+	maxNotAfter := time.Date(9999, 12, 31, 0, 0, 0, 0, time.UTC)
+	if certNotAfter.After(maxNotAfter) {
+		certNotAfter = maxNotAfter
+	}
```

Comment:
Is this check necessary? Go's `time.Time` can store values much higher than this, and timestamps in certificates are just encoded as strings and so do not have true max values.

Reply:
I'm afraid it is. If you go one second further than 9999-12-31 23:59:59 you'll overflow the "notAfter" field, which is a "GeneralizedTime":

> 4.1.2.5.2.  GeneralizedTime
> 
>    The generalized time type, GeneralizedTime, is a standard ASN.1 type
>    for variable precision representation of time.  Optionally, the
>    GeneralizedTime field can include a representation of the time
>    differential between local and Greenwich Mean Time.
> 
>    For the purposes of this profile, GeneralizedTime values MUST be
>    expressed in Greenwich Mean Time (Zulu) and MUST include seconds
>    (**i.e., times are YYYYMMDDHHMMSSZ**), even where the number of seconds
>    is zero.  GeneralizedTime values MUST NOT include fractional seconds.

The reason why I build this check in the first place is because otherwise Pebble would fail to issue the certificate and my certbot would just keep polling Pebble without any chance of success.

Reply:
By the way, I'm not sure if this is still necessary since we're using `time.Duration()` with `time.Add()`:

> A Duration represents the elapsed time between two instants as an int64 nanosecond count. The representation limits the largest representable duration to approximately 290 years. ([type Duration](https://pkg.go.dev/time#Duration))